### PR TITLE
Adding ability to provide custom editors

### PR DESCRIPTION
**Pull request by abhirao**
Addresses #162

- This isn't ready to merge by any means. I got something working so I could wrap my brain around the problem. Now I'm wondering what the best interface could be. I believe I have a change that may not be backwards compatible - fieldValue is now being passed from TableBody to TableEditColumn as an attribute.

- This is a working example:

cell_editing

### CODE REVIEW

**Comment by AllenFang** [comment]
@abhirao, it's seems look like good at first sight. Maybe can released on ```v3.0.0```. How about it? or did you want to released in near future?

File: src/TableEditColumn.js [hunk 1]
```diff
@@ -116,7 +126,12 @@ TableEditColumn.propTypes = {
   blurToSave: PropTypes.bool,
   editable: PropTypes.oneOfType([ PropTypes.bool, PropTypes.object ]),
   format: PropTypes.oneOfType([ PropTypes.bool, PropTypes.func ]),
-  children: PropTypes.node
+  fieldValue: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.bool,
+    PropTypes.number,
+    PropTypes.object
+  ])
```
Comment (abhirao) [inline src/TableEditColumn.js]: @AllenFang I'm not sure what the impact is of changing the fieldValue (which used to come in as children) from node to the 4 types I have listed.

**Comment by abhirao** [comment]
@AllenFang Thank you very much for creating this library! Sorry I was traveling last week and didn't get a chance to respond to you.

I'm not sure what the timeline is for v3.0.0 but I think our team would like the custom cell editing sooner. I can definitely help with this.

**Comment by AllenFang** [comment]
@abhirao, that's awesome, I'll expect your contribution:) So I just keep this PR open and wait your confirm for completion.

BTW, if you complete this PR and if I almost finish ```v3.0.0```, I'll release them together.

**Comment by abhirao** [comment]
I analyzed the backwards compatibility of this PR and I think it should be fine.  The only prop type that is changing is for editable columns for the field values.

My current implementation of custom editors is following a different pattern than custom filters. I'm going to do a little refactor to make the customEditor work the same way as filter.

**Comment by abhirao** [comment]
@AllenFang I've updated the interface for my custom editor implementation to match the custom filter interface.

Custom Editor:

`<TableHeaderColumn
            dataField='price'
            dataFormat={ priceFormatter }
            customEditor={ { getElement: createPriceEditor, customEditorParameters: { currencies: currencies } } }>
            Product Price
          </TableHeaderColumn>`

Custom Filter:

`<TableHeaderColumn dataField='isInStock' filter={ { type: 'CustomFilter', getElement: getCustomFilter, customFilterParameters: { textOK: 'yes', textNOK: 'no' } } }>Product Is In Stock</TableHeaderColumn>`

I think it's good to go. What do you think? Let me know if/how I can help with documentation.

**Comment by AllenFang** [comment]
@abhirao, that's awesome :) I'll spend some time to review this PR. But need some time, it's busy in these day ;(

Anyway, you did a great job and thanks you!

**Comment by abhirao** [comment]
@AllenFang You're welcome.

 Thank you very much for all you do keep this project going 🙇 .

**Comment by AllenFang** [comment]
Just a passion on it to keep me to contribute this repo. I'm busy in the recent, but I think I can back in next month! Anyway, if u r interesting for contributing this repo at long term, welcome to discuss with me :)

**Comment by AllenFang** [comment]
@abhirao, I need to say thanks agin. This PR has been released on ```v2.4.0```. Thanks you :) That's awesome!

**Pull request merged**